apps/web/lib/getMainDomainOrgRedirect.ts

```diff
@@ -0,0 +1,33 @@
+import type { IncomingMessage } from "http";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import logger from "@calcom/lib/logger";
+
+const log = logger.getSubLogger({ prefix: ["lib", "getMainDomainOrgRedirect"] });
+export const getMainDomainOrgRedirect = (req: IncomingMessage | undefined) => {
+  if (!req) {
+    return null;
+  }
+
+  const protocol = WEBAPP_URL.startsWith("https://") ? "https" : "http";
+  const originalUrl = new URL(req.url!, `${protocol}://${req.headers.host}`);
+
+  const domains = originalUrl.hostname.split(".");
```

Comment:
Let's be clear with naming, hostname split doesn't give domains. May be we can call it, domainParts.